core/src/gl/render_device.cpp

```diff
@@ -59,3 +60,34 @@ bool FramebufferDesc::FramebufferTarget::isSet() const
 {
     return mSet;
 }
+
+impl::PipelinedTimer::PipelinedTimer(RenderDevice* rd)
+    : mRd(rd)
```

Comment:
```suggestion
impl::PipelinedTimer::PipelinedTimer(RenderDevice& rd)
    : mRd(rd)
```
Store a reference instead: it can never be null anyway, better make it clear.